### PR TITLE
Health checks flow

### PR DESCRIPTION
**Pull request by Preeticp**
This PR adds content for Health checks via Developer perspective flow.
This is relevant to the 4.5 version of docs.
It has been reviewed by SMEs, QE, and peer reviewed by the Devtools docs team

### CODE REVIEW

File: modules/odc-importing-codebase-from-git-to-create-application.adoc [hunk 1]
```diff
@@ -56,8 +57,22 @@ Click the *Routing* link to:
...
+* Click *Add Liveness Probe*, if required, modify the parameters to check if a container is still running, and select the check mark to add the probe.
+* Click *Add Startup Probe*, if required, modify the parameters to check if the application within the container has started, and select the check mark to add the probe.
++
+For each of the probes, you can specify the request type - *HTTP GET*, *Container Command*, or *TCP Socket*,  from the drop-down list. The form changes as per the selected request type. You can then modify the default values for the other parameters, such as the success and failure thresholds for the probe, number of seconds before performing the first probe after the container starts, frequency of the probe, and the timeout value.
+
+
+Build Configuration and Deployment::
+Click the *Build Configuration* and *Deployment* links to see the respective configuration options. Some of the options are selected by default; you can customize them further by adding the necessary triggers and environment variables.
```
Comment (boczkowska) [inline modules/odc-importing-codebase-from-git-to-create-application.adoc:73]: _Some of the_ can be safely replace by _some_

File: modules/odc-adding-health-checks.adoc [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+= Adding health checks
+
+You can use the *Topology* view to add health checks to your deployed application.
+
+.Procedure
+. In the *Topology* view, click on the application node to see the side panel. If the container does not have health checks added to ensure the smooth running of your application, a *Health Checks* notification is displayed with a link to add health checks.
+. In the displayed notification, click the *Add Health Checks* link.
+. Alternatively, you can also click the *Actions* drop-down list and select *Add Health Checks*. Note that if the container already has health checks, you will see the *Edit Health Checks* option instead of the add option.
+. In the *Add Health Checks* form, if you have deployed multiple containers, use the *Container* drop-down list to ensure that the appropriate container is selected.
+. Click the add probe links to add the required health checks. Default data for the health checks is pre-populated. You can add the probes with the default data or further customize the values and then add them. For example, to add a Readiness probe that checks if your container is ready to handle requests:
```
Comment (boczkowska) [inline modules/odc-adding-health-checks.adoc:15]: I believe IBM advises against hyphening prefixes. 
I couldn't find "prepopulate" in their examples but according to quick search "prepopulate" is used more in the books/documentation in general.

File: modules/odc-editing-health-checks.adoc [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+You can use the *Topology* view to edit health checks added to your application, modify them, or add more health checks.
+
+.Procedure
+. In the *Topology* view, right-click your application and select *Edit Health Checks*. Alternatively, in the side panel, click the *Actions* drop-down list and select *Edit Health Checks*.
+. In the *Edit Health Checks* page:
+
+* To remove a previously added health probe, use the minus sign adjoining it.
+* To edit the parameters of an existing probe:
++
+.. Click the *Edit Probe* link next to a previously added probe or click on the probe to see the parameters for the probe.
```
Comment (boczkowska) [inline modules/odc-editing-health-checks.adoc:17]: click on -> click ?

Not sure about this one.

File: modules/odc-editing-health-checks.adoc [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+[id="odc-editing-health-checks"]
+= Editing health checks
+
+You can use the *Topology* view to edit health checks added to your application, modify them, or add more health checks.
+
+.Procedure
+. In the *Topology* view, right-click your application and select *Edit Health Checks*. Alternatively, in the side panel, click the *Actions* drop-down list and select *Edit Health Checks*.
+. In the *Edit Health Checks* page:
+
+* To remove a previously added health probe, use the minus sign adjoining it.
```
Comment (boczkowska) [inline modules/odc-editing-health-checks.adoc:14]: use the minus sign -> click the minus sign ?

File: modules/odc-editing-health-checks.adoc (hunk 3 above)
Comment (Preeticp) [reply modules/odc-editing-health-checks.adoc:17]: Click, as you suggest, would be correct. But the developer asked me to remove that part, so have taken that part out.

**Review by vikram-raj — COMMENTED** [review]
Add and Edit Health checks content LGTM!. 
Thanks @Preeticp

File: modules/odc-adding-health-checks.adoc [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+. In the displayed notification, click the *Add Health Checks* link.
+. Alternatively, you can also click the *Actions* drop-down list and select *Add Health Checks*. Note that if the container already has health checks, you will see the *Edit Health Checks* option instead of the add option.
+. In the *Add Health Checks* form, if you have deployed multiple containers, use the *Container* drop-down list to ensure that the appropriate container is selected.
+. Click the add probe links to add the required health checks. Default data for the health checks is prepopulated. You can add the probes with the default data or further customize the values and then add them. For example, to add a Readiness probe that checks if your container is ready to handle requests:
+.. Click *Add Readiness Probe*, to see a form containing the parameters for the probe.
+.. Click the *Type* drop-down list to select the type of checks you want to add. For example, in this case, select *Container Command* to select the command that will be executed inside the container.
```
Comment (debsmita1) [inline modules/odc-adding-health-checks.adoc:21]:
```suggestion
.. Click the *Type* drop-down list to select the request type you want to add. For example, in this case, select *Container Command* to select the command that will be executed inside the container.
```

**Review by debsmita1 — CHANGES_REQUESTED** [review]

File: modules/odc-adding-health-checks.adoc [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+* You have created and deployed an application on {product-title} using the *Developer* perspective.
+
+.Procedure
+. In the *Topology* view, click on the application node to see the side panel. If the container does not have health checks added to ensure the smooth running of your application, a *Health Checks* notification is displayed with a link to add health checks.
+. In the displayed notification, click the *Add Health Checks* link.
+. Alternatively, you can also click the *Actions* drop-down list and select *Add Health Checks*. Note that if the container already has health checks, you will see the *Edit Health Checks* option instead of the add option.
+. In the *Add Health Checks* form, if you have deployed multiple containers, use the *Container* drop-down list to ensure that the appropriate container is selected.
+. Click the add probe links to add the required health checks. Default data for the health checks is prepopulated. You can add the probes with the default data or further customize the values and then add them. For example, to add a Readiness probe that checks if your container is ready to handle requests:
```
Comment (ahardin-rh) [inline modules/odc-adding-health-checks.adoc:19]: should add probes be formatted as follows? Not sure if it's a UI item.

    *Add Probes*

File: modules/odc-adding-health-checks.adoc [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+.. Click *Add Readiness Probe*, to see a form containing the parameters for the probe.
+.. Click the *Type* drop-down list to select the type of checks you want to add. For example, in this case, select *Container Command* to select the command that will be executed inside the container.
+.. In the *Command* field, add an argument `cat`, similarly, you can add multiple arguments for the check, for example, add another argument `/tmp/healthy`.
+.. Retain or modify the default values for the other parameters as required, and click the check mark at the bottom of the form. The *Readiness Probe Added* message is displayed.
+. Click *Add* to add the health check. You are redirected to the *Topology* view and the container is restarted.
+. In the side panel, verify that the probes have been added by clicking on the deployed pod under the *Pods* section.
```
Comment (ahardin-rh) [inline modules/odc-adding-health-checks.adoc:25]: s/pod/Pod

File: modules/odc-editing-health-checks.adoc [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
++
+.. Click the *Edit Probe* link next to a previously added probe to see the parameters for the probe.
+.. Modify the parameters as required, and click the check mark to save your changes.
++
+* To add a new health probe, in addition to existing health checks, click the add probe links. For example, to add a Liveness probe that checks if your container is running:
++
+.. Click *Add Liveness Probe*, to see a form containing the parameters for the probe.
+.. Edit the probe parameters as required, and click the check mark at the bottom of the form. The *Liveness Probe Added* message is displayed.
+. Click *Save* to save your modifications and add the additional probes to your container. You are redirected to the *Topology* view.
+. In the side panel, verify that the probes have been added by clicking on the deployed pod under the *Pods* section.
```
Comment (ahardin-rh) [inline modules/odc-editing-health-checks.adoc:30]: s/pod/Pod

File: modules/odc-importing-codebase-from-git-to-create-application.adoc [hunk 9]
```diff
@@ -56,8 +57,22 @@ Click the *Routing* link to:
 +
 For serverless applications, the Knative Service manages all the routing options above. However, you can customize the target port for traffic, if required. If the target port is not specified, the default port of `8080` is used.
 
-Build and Deployment Configuration::
-Click the *Build Configuration* and *Deployment Configuration* links to see the respective configuration options. Some of the options are selected by default; you can customize them further by adding the necessary triggers and environment variables. For serverless applications, the *Deployment Configuration* option is not displayed as the Knative configuration resource maintains the desired state for your deployment instead of a DeploymentConfig.
+Health Checks::
+Click the *Health Checks* link to add Readiness, Liveness, and Startup probes to your application. All the probes have prepopulated default data, you can add the probes with the default data or customize it as required.
```
Comment (ahardin-rh) [inline modules/odc-importing-codebase-from-git-to-create-application.adoc:61]: s/,you can/; you can

**Comment by ahardin-rh** [comment]
Just some minor comments from me! Once the changes are squashed into one commit, we should be good to go! :smile:

File: modules/odc-editing-health-checks.adoc [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+* You have added health checks to your application.
+
+.Procedure
+. In the *Topology* view, right-click your application and select *Edit Health Checks*. Alternatively, in the side panel, click the *Actions* drop-down list and select *Edit Health Checks*.
+. In the *Edit Health Checks* page:
+
+* To remove a previously added health probe, click the minus sign adjoining it.
+* To edit the parameters of an existing probe:
++
+.. Click the *Edit Probe* link next to a previously added probe to see the parameters for the probe.
```
Comment (gajanan194) [inline modules/odc-editing-health-checks.adoc:22]: Edit Probe is near to the title of the Probe.

**Review by gajanan-more — CHANGES_REQUESTED** [review]

File: modules/odc-editing-health-checks.adoc (hunk 10 above)
Comment (Preeticp) [reply modules/odc-editing-health-checks.adoc:22]: Imho, the current description is correct and the UI is clear enough that the user will not get confused. So I will not make any change to that if you don't mind.

File: modules/odc-adding-health-checks.adoc (hunk 6 above)
Comment (Preeticp) [reply modules/odc-adding-health-checks.adoc:19]: So the UI actually says *Add <name of the probe> Probe* which I mention in the example for Readiness probe. Have reworded it like this: Click the required health probe links to add them to the container.

**Comment by Preeticp** [comment]
@ahardin-rh I have fixed all the comments, and squashed the commits, please merge.

**Pull request merged**